src/test/java/org/opensearch/knn/index/query/KNNQueryBuilderTests.java

```diff
@@ -485,6 +486,7 @@ public void testDoToQuery_Normal() throws Exception {
         assertEquals(knnQueryBuilder.vector(), query.getQueryVector());
     }
 
+    @SneakyThrows
```

Comment:
Seems like we skipped adding unit test, any particular reason? if not lets add it

Reply:
+1

Reply:
we cannot test real behavior of the rewrite for the query in the unit test, only in integ tests. We already have unit tests for general logic for cases when filter is null and not null.

Reply:
added unit tests that follow this logic, just using mocked context objects